### ISSUE TITLE
GPU Training

### ISSUE THREAD

**Issue by ps793** [ostamand/tensorflow-tabnet#7]
Does this repo support GPU training?

**Comment by trulicup** [comment]
Yeah it does. I tested it.

**Comment by ps793** [comment]
I double check with the log file, it does support GPU. however, when I run the experiments, i found that training time is not stable. as shown below:
"
Epoch 00010: LearningRateScheduler reducing learning rate to 0.01.
Epoch 10/60
Epoch 00010: val_loss improved from 17.12721 to 16.89577, saving model to /opt/ml/model/
1231/1231 - 93s - loss: 16.9219 - rmse: 22.7585 - val_loss: 16.8958 - val_rmse: 22.7614

Epoch 00011: LearningRateScheduler reducing learning rate to 0.01.
Epoch 11/60
Epoch 00011: val_loss did not improve from 16.89577
1231/1231 - 58s - loss: 16.7816 - rmse: 22.5957 - val_loss: 17.6749 - val_rmse: 23.6871
"
it seems like longer training time will benefit to have better performance. (val loss) 

I assumed longer one will use CPU and shorter one use GPU so that I raise this issue. But it seems like they are all in GPU.

Does anyone know why the training time changes?

**Comment by ostamand** [comment]
@ps793 This is probably caused by the dataset `tf.data.experimental.AUTOTUNE` setting. It optimizes the parallelization during training but it does generate differences in training speed between the runs especially for a low number of epochs.

**Comment by ostamand** [comment]
Let me know if I did not understand the issue correctly.

**Comment by ps793** [comment]
@ostamand do you have similar issue when you implement tabnet? based on what i find is that, shorter time running may occur more frequently in the later epochs, like 50/60 to 55/60. i guess when there are more overfiting, the training time is shorter? i also test another version of tabnet tf2.0, they have the same issue.

**Comment by ostamand** [comment]
In my experience it tend to depends a lot on the hardware you are on and/or the version of TF used.
